utils/utils.go

```diff
@@ -38,15 +37,15 @@ var (
 //
 
 // CheckStakeWeightExceedsThreshold returns true if the accumulated signature weight is at least [quorumNum]/[quorumDen] of [totalWeight].
-func CheckStakeWeightExceedsThreshold(accumulatedSignatureWeight *big.Int, totalWeight uint64) bool {
+func CheckStakeWeightExceedsThreshold(accumulatedSignatureWeight *big.Int, totalWeight uint64, quorumNumerator uint64, quorumDenominator uint64) bool {
 	if accumulatedSignatureWeight == nil {
 		return false
 	}
 
 	// Verifies that quorumNum * totalWeight <= quorumDen * sigWeight
-	scaledTotalWeight := new(big.Int).SetUint64(totalWeight)
-	scaledTotalWeight.Mul(scaledTotalWeight, new(big.Int).SetUint64(DefaultQuorumNumerator))
-	scaledSigWeight := new(big.Int).Mul(accumulatedSignatureWeight, new(big.Int).SetUint64(DefaultQuorumDenominator))
+	totalWeightBI := new(big.Int).SetUint64(totalWeight)
```

Comment:
Is there a specific reason to introduce the new variable `totalWeightBI`? When compared to operating on `scaledTotalWeight` in place, there's an additional allocation incurred.

Reply:
I made this change to make the code a little more clear. Previously we were assigning `scaledTotalWeight := new(big.Int).SetUint64(totalWeight)`, but it didn't actually represent the scaled total weight until the operation on the next line.

If we think the performance hit of the allocation is a big deal, we can change it back. We could also eliminate another allocation on line 48 by re-using `scaledSigWeight`.

Reply:
I think the extra allocation for clarify in variable naming is okay